lib/rubygems/specification.rb

```diff
@@ -2711,11 +2711,18 @@ def validate packaging = true
         raise Gem::InvalidSpecificationException,
           "each license must be 64 characters or less"
       end
+
+      if !Gem::SPDX::IDENTIFIERS.include?(license)
+        warning <<-warning % license
+WARNING: license value '%s' is invalid.  Use a license identifier from:
+http://spdx.org/licenses
+        warning
```

Comment:
@bf4, I've gone from a frozen template to an interpolated string that shows both the invalid value and the correct value for nonstandard licenses.